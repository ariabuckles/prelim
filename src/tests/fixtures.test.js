const fs = require('fs');
const path = require('path');
const babel = require('@babel/core');
const plugin = require('../plugin').default;

const fixturePath = path.join(__dirname, 'fixtures');

const fixtures = fs
  .readdirSync(fixturePath)
  .filter((fileName) => /\.js$/.test(fileName))
  .map((fileName) => path.join(fixturePath, fileName));

describe('fixtures', () => {
  for (let fixture of fixtures) {
    it(`should optimize ${path.basename(fixture)}`, () => {
      let { code } = babel.transformFileSync(fixture, { plugins: [plugin] });
      expect(code).toMatchSnapshot();
    });
  }
});
