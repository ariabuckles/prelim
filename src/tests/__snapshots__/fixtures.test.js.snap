// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fixtures should optimize simple-variable.js 1`] = `
"let b = 6;
console.log(b);"
`;

exports[`fixtures should optimize unused-properties.js 1`] = `
"let object1 = {
  a: console.log(1),
  b: 2
};
console.log(object1.b);"
`;
